### PR TITLE
Introduce `FluxCollect` Refaster rule

### PR DESCRIPTION
**Pull request by rickie**
As suggested by @Stephan202 in [this PR](https://github.com/PicnicSupermarket/error-prone-support/pull/703#issuecomment-1613280549).

### CODE REVIEW

**Comment by rickie** [comment]
Ofcourse, how could I forget about this one....

> Warning:  /home/runner/work/error-prone-support/error-prone-support/error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java:[1187,44] [NonEmptyMono] Avoid vacuous operations on known non-empty `Mono`s
    (see https://error-prone.picnic.tech/bugpatterns/NonEmptyMono)
  Did you mean 'return flux.collect(collector);'?

It's funny that I got a few questions during some of the presentations about how to make sure that you do not have overlapping rules. I'm glad that I can explain the epic self-check to those people 😄. 

Will close this.

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 1]
```diff
@@ -1180,6 +1180,19 @@ Flux<T> after(Flux<T> flux, Predicate<? super T> predicate, Comparator<? super T
     }
   }
 
+  /* Don't unnecessarily invoke {@link Mono#single()} as {@link Flux#collect(Collector)} always yields one item. */
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java:1183]: Javadoc :)

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java [hunk 2]
```diff
@@ -1180,6 +1180,19 @@ Flux<T> after(Flux<T> flux, Predicate<? super T> predicate, Comparator<? super T
...
+  static final class FluxCollect<T, R> {
+    @BeforeTemplate
+    Mono<R> before(Flux<T> flux, Collector<T, ?, R> collector) {
```
Comment (Stephan202) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java:1186]: The collector accepts a `super T`.

**Review by Stephan202 — COMMENTED** [review]
Rebased and added a commit. Suggested commit message:
```
Introduce `FluxCollect` Refaster rule (#713)
```

**Comment by Stephan202** [comment]
Doh, I had an old tab open :stuck_out_tongue:

**Comment by rickie** [comment]
Thanks for the review though @Stephan202 😉.

**Pull request closed without merging**